src/Uno.UI/UI/Xaml/Controls/TextBox/SinglelineTextBoxView.iOS.cs

```diff
@@ -99,29 +99,40 @@ private void Initialize()
 
 		partial void OnLoadedPartial()
 		{
-			this.EditingChanged += OnEditingChanged;
-			this.EditingDidEnd += OnEditingChanged;
+			SubscribeEditingEvents();
 		}
 
 		partial void OnUnloadedPartial()
 		{
-			this.EditingChanged -= OnEditingChanged;
-			this.EditingDidEnd -= OnEditingChanged;
+			UnsubscribeEditingEvents();
 		}
 
 		//Forces the secure UITextField to maintain its current value upon regaining focus
 		public override bool BecomeFirstResponder()
 		{
 			var result = base.BecomeFirstResponder();
 
-			if (SecureTextEntry)
+			if (SecureTextEntry && !string.IsNullOrEmpty(Text))
 			{
+				UnsubscribeEditingEvents();
+
 				var text = Text;
-				Text = string.Empty;
+				UpdatePasswordText(string.Empty);
 				InsertText(text);
+				UpdatePasswordText(text);
+
+				SubscribeEditingEvents();
 			}
```

Comment:
wouldnt it be better to use a flag to guard the handlers, instead of un- and re-registering the events?